### DISCUSSION TITLE
Does pulumi support partial deployments?

### DISCUSSION THREAD

**Discussion by kenseii** [pulumi/pulumi#10044] [Pulumi in Practice] [answered]
Hello,
Trying out pulumi for a new project and as it might have multiple components in it, i am wondering if its possible to deploy a specific part of the project. for example `pulumi up filename.extension`

I tried looking around but the docs don't mention anything about partial deployments

Thank you

**Comment by jaxxstorm** [comment, marked as answer]
HI @kenseii - thanks for opening this discussion!

Yes, Pulumi can do partial deployments, but not against a single file like your example.

Pulumi can do _targeted_ deployments of existing resources using `--target` flag on a urn, which is a resource identifier. That resources must have already been created until https://github.com/pulumi/pulumi/issues/8956 is resolved.

If you want to do deployments of only a few resources, generally it's recommended to break your projects and stacks apart using a [micro-stacks](https://www.pulumi.com/docs/guides/organizing-projects-stacks/#micro-stacks) model which will get you the functionality you need.

I hope this helps, please feel free to follow up with any additional questions

**Comment by kenseii** [comment]
@jaxxstorm Thank you very much, i think i can achieve that using the `--target` flag.
also i found that i can get the URNs of the resources by running `pulumi stack --show-urns`

I didn’t know much about the way pulumi handles URN but after reading this it was more clear.
https://www.pulumi.com/docs/intro/concepts/resources/names/#urns

**Comment by zhukovgreen** [comment]
When you are not interested in particular state during a partial deployment and just need to split the deployment into N pieces you can go with the following approach:
```
❯ timeout --foreground -k 5m -s INT 55m pulumi up --yes
```
After 55 mins  it sends SIGINT and after 5 mins after SIGINT it will send SIGKILL

SIGINT is equal to user pressing ^C and pulumi gracefully cancel not yet running tasks and waits for already running tasks to accomplish.

Then simple rerunning the same command starts from the point where it left of during the previous command run
